epochX/cudacpp/CODEGEN/PLUGIN/CUDACPP_SA_OUTPUT/model_handling.py

```diff
@@ -823,15 +823,30 @@ def super_generate_parameters_class_files(self):
         """Create the content of the Parameters_model.h and .cc files"""
         # First of all, identify which extra independent parameters must be made available through CPU static and GPU constant memory in BSM models
         # because they are used in the event by event calculation of alphaS-dependent couplings
-        # WARNING! This is only implemented and has only been tested so far for real parameters (complex parameters need twice the storage)
         bsmparam_indep_real_used = []
-        if self.model_name[:2] != 'sm' :
-            for param in self.params_indep:
-                if param.type == 'real':
-                    for coup in self.coups_dep.values():                
-                        if param.name in coup.expr:
-                            bsmparam_indep_real_used.append( param.name )
-            bsmparam_indep_real_used = set( bsmparam_indep_real_used ) 
+        bsmparam_indep_complex_used = []
+        for param in self.params_indep: # NB this is now done also for 'sm' processes (no check on model name, see PR #824)
+            if param.name == 'mdl_complexi' : continue
+            if param.name == 'aS' : continue
+            # Add BSM parameters which are needed to compute dependent couplings
+            # Note: this seemed enough to fix SUSY processes, but not EFT processes
+            for coupdep in self.coups_dep.values():
+                if param.name in coupdep.expr:
```

Comment:
things to check, do we need regular expression here to avoid overlap
but I guess 
mdl_g
will be return true for 
mdl_g1 as expression

But likely not critical since just too much
-> Andrea answer (thinks this is include in a next PR -> I will check)

Reply:
Thanks @oliviermattelaer 

I tried to use ' '+param.name+' ' instead, but this is not enough. See cfd56a4ff in PR #832. 
I understand this is not ideal, maybe something to be addressed in mg5amcnlo#90?
Not much else I can do here I think I am afraid

Reply:
I have also added a comment in a patch in #832